crates/sui-faucet/src/server.rs

```diff
@@ -53,6 +54,24 @@ pub async fn start_faucet(
         ..
     } = app_state.config;
 
+    // Max request per second as bursts, and replenishes one after 3600 seconds based on Peer IP.
+    let governor_conf = Arc::new(
+        GovernorConfigBuilder::default()
+            .per_second(3600)
+            .burst_size(max_request_per_second as u32)
+            .finish()
+            .unwrap(),
+    );
+
+    let governor_limiter = governor_conf.limiter().clone();
+    let interval = Duration::from_secs(60);
+    // a separate background task to clean up
+    std::thread::spawn(move || loop {
```

Comment:
You're mixing thread-based scheduling (`thread::spawn`) and `tokio` based scheduling (`tokio::time::sleep`) here.  Stick to tokio. Look at the WAL cleanup task below for reference.

Reply:
Ah yes, sorry. It was late and only fixed it in one place (the sleep one) :D